File: samples/android/AdUnitsSample/src/main/java/com/facebook/samples/AdUnitsSample/SampleListActivity.java
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 * All rights reserved.
 *
 * This source code is licensed under the license found in the
 * LICENSE file in the root directory of this source tree.
 */

package com.facebook.samples.AdUnitsSample;

import android.app.ListActivity;
import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;
import android.view.View;
import android.widget.AdapterView;
import androidx.fragment.app.Fragment;
import com.facebook.samples.AdUnitsSample.adapters.SampleAdapter;
import com.facebook.samples.ads.debugsettings.DebugSettingsActivity;

/** A simple {@link Fragment} subclass. */
public class SampleListActivity extends ListActivity {

  private static final String TAG = SampleListActivity.class.getSimpleName();

  @Override
  protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);

    setTitle(R.string.select_sample);

    final SampleAdapter adapter = new SampleAdapter(SampleListActivity.this);

    // Bind to our new adapter.
    setListAdapter(adapter);

    getListView()
        .setOnItemClickListener(
            new AdapterView.OnItemClickListener() {
              @Override
              public void onItemClick(AdapterView<?> parent, View view, int position, long id) {

                Log.e(TAG, "List item clicked: " + position);
                SampleAdapter.Item item = (SampleAdapter.Item) adapter.getItem(position);

                String sampleName = item.getTitle();
                AdUnitsSampleType type = AdUnitsSampleType.getSampleTypeFromName(sampleName);
                if (type != null) {
                  Intent intent = new Intent(SampleListActivity.this, AdUnitsSampleActivity.class);
                  intent.putExtra(AdUnitsSampleActivity.SAMPLE_TYPE, type.getName());
                  startActivity(intent);
                }
              }
            });
  }

  @Override
  public boolean onCreateOptionsMenu(Menu menu) {
    MenuInflater inflater = getMenuInflater();
    inflater.inflate(R.menu.ad_units_sample_menu, menu);
    return true;
  }

  @Override
  public boolean onOptionsItemSelected(MenuItem item) {
    int i = item.getItemId();
    if (i == R.id.debug_settings) {
      startActivity(new Intent(getApplicationContext(), DebugSettingsActivity.class));
      return true;
    }

    return super.onOptionsItemSelected(item);
  }
}

File: samples/android/KotlinAdUnitsSample/src/main/java/com/facebook/samples/adunitssamplekotlin/SampleListActivity.kt
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 * All rights reserved.
 *
 * This source code is licensed under the license found in the
 * LICENSE file in the root directory of this source tree.
 */

package com.facebook.samples.adunitssamplekotlin

import android.content.Intent
import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import androidx.appcompat.app.AppCompatActivity
import com.facebook.samples.ads.debugsettings.DebugSettingsActivity
import com.facebook.samples.adunitssamplekotlin.fragments.SampleListFragment

class SampleListActivity : AppCompatActivity() {

  companion object {
    val TAG = SampleListActivity::class.java.simpleName
  }

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)

    setContentView(R.layout.activity_sample_list)

    supportFragmentManager
        .beginTransaction()
        .add(R.id.list_fragment_container, SampleListFragment())
        .commit()
  }

  override fun onCreateOptionsMenu(menu: Menu): Boolean {
    menuInflater.inflate(R.menu.ad_units_sample_menu, menu)
    return true
  }

  override fun onOptionsItemSelected(item: MenuItem): Boolean {
    if (item.itemId == R.id.debug_settings) {
      // Start debug settings
      startActivity(Intent(this, DebugSettingsActivity::class.java))
      return true
    }

    return super.onOptionsItemSelected(item)
  }
}
